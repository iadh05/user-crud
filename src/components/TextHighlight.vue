<template>
    <span v-html="parsedText" class="text-highlight"></span>
</template>

<script>
export default {
    name: "TextHighlight",
    props: ["text", "search"],
    computed: {
        parsedText() {
            if (!this.search || !this.text) return this.text;
            const text = this.text.toString();

            const regex = new RegExp(this.search, "gi");
            const found = text.match(regex);
            if (!found) return text;
            const arr = text.split(regex);
            let result = "";
            for (const [index, item] of arr.entries()) {
                result +=
                    item +
                    (found[index] ? `<strong>${found[index]}</strong>` : "");
            }
            return result;
        },
    },
};
</script>

<style lang="scss">
.text-highlight {
    strong {
        font-weight: 900;
    }
}
</style>
