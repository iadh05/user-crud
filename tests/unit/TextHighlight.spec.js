import Vue from 'vue'
import { mount, createLocalVue } from '@vue/test-utils'
import TextHighlight from '@/components/TextHighlight.vue'

import Vuetify from 'vuetify'
const localVue = createLocalVue()

Vue.use(Vuetify);
describe('TextHighlight.vue', () => {
    it('should show TextHighlight', () => {
        const wrapper = mount(TextHighlight, {

            localVue,
            vuetify: new Vuetify,


        })

        expect(wrapper.html()).toMatchSnapshot()
    })
    it('should show TextHighlight', () => {
        const wrapper = mount(TextHighlight, {

            localVue,
            vuetify: new Vuetify,

            propsData: {
                text: "Jhon",
                search: "J"
            }
        })
        const highlighitedText = wrapper.find("[data-cy='highlighitedText']")
        expect(highlighitedText.exists()).toBe(true)
        expect(highlighitedText.text()).toBe("J")
    })
})