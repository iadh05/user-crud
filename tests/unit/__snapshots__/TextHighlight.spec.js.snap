// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHighlight.vue should show TextHighlight 1`] = `<span class="text-highlight"></span>`;
